include/nuttx/spinlock.h

```diff
@@ -30,7 +30,11 @@
 #include <sys/types.h>
 #include <stdint.h>
 
-#ifdef CONFIG_SPINLOCK
+#ifndef CONFIG_SPINLOCK
```

Comment:
I'm trying to understand the outcome of this change when `CONFIG_SPINLOCK` is not defined.
From what I understood, `&priv->lock` should always evaluate to `NULL` when `CONFIG_SPINLOCK` is not defined.
I've tried to isolate this behavior, but I couldn't achieve the same result.

https://godbolt.org/z/467afd

Did I do something wrong in my test?

Reply:
Hi @gustavonihei,

Actually, when CONFIG_SPINLOCK is not defined CONFIG_SMP is not defined either.
So, finally, spin_lock_irqsave(l) will be just up_irq_save().

```
include/nuttx/irq.h

...
#if defined(CONFIG_SMP)                                                                                                                                                
irqstate_t spin_lock_irqsave(spinlock_t *lock);                                                                                                                        
#else                                                                                                                                                                  
#  define spin_lock_irqsave(l) up_irq_save()                                                                                                                           
#endif  
...
```

Reply:
Thanks for the clarification, @masayuki2009!